pkg/driver/driver.go

```diff
@@ -45,7 +45,7 @@ func New(driverName, nodeID, endpoint, dataRoot, tmpfsSize string) (*Driver, err
 
 	mntPoint, err := util.IsLikelyMountPoint(dataRoot)
 	if os.IsNotExist(err) {
-		if err = os.MkdirAll(dataRoot, 0700); err != nil {
+		if err = os.MkdirAll(dataRoot, (0700 | os.ModeSticky)); err != nil {
```

Comment:
Add an explanatory comment about why sticky bit is needed here,
And in my mind, sticky bit only makes sense when a directory is group or world writable, because it prevents other users from writing or deleting other users files and sub-directories.

Also, if this is a mountpoint, then it will be the permissions in the filesystem mounted here that are important and this directory will not be populated while it is mounted.